spec/models/campaign_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe Campaign, type: :model do
+  let(:campaign) { build(:campaign) }
+
+  context 'when title field is empty' do
+    it { is_expected.to_not allow_value('').for(:title) }
+  end
+
+  context 'when title field is too short' do
+    it { is_expected.to validate_length_of(:title).is_at_least(3) }
+  end
+
+  context 'when title field is too long' do
+    it { is_expected.to validate_length_of(:title).is_at_most(30) }
```

Comment:
you can have only one test for title instead of three
one for description


read the documentation and rewrite all of the tests https://github.com/thoughtbot/shoulda-matchers